src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -1065,6 +1065,9 @@ private void checkWritebackEnabled() {
      * the field locations in a form usable by {@link #getInt(Object,long)}.
      * Therefore, code which will be ported to such JVMs on 64-bit platforms
      * must preserve all bits of static field offsets.
+     *
+     * @throws NullPointerException if the field is {@code null}
+     * @throws IllegalArgumentException if the field is static
```

Comment:
Maybe these checks could be done in **Java** like:
```java
// implicit null check:
if ((f.getModifiers() & ACC_STATIC) != 0) {
	throw new IllegalArgumentException("Field is static");
}
```
and for the `staticField(Offset/Base)` methods:
```java
// implicit null check:
if ((f.getModifiers() & ACC_STATIC) == 0) {
	throw new IllegalArgumentException("Field is not static");
}
```

Reply:
Don't think anyone is willing to change code here...

Reply:
As I commented on the C++ change, I think it's better to put the check in Java rather than C++.

The behavioral results are equivalent, right?  It is fine to upgrade the InternalError to IAE; InternalError is by definition a poorly-defined state, and IAE is a less poorly defined way to report it.  Changing that path to throw IAE is better, since it aligns with other field-query APIs.  Because we control use of Unsafe (in this package) we can make small changes like this and just adjust our own code if something breaks.

Error checks and validations in Java code will be 10x clearer (more maintainable, scrutable to JITs) than logic in the C++ code.